src/lib/y2partitioner/widgets/overview.rb

```diff
@@ -76,8 +122,11 @@ def disks_items
 
       def partition_items(disk)
         disk.partitions.map do |partition|
+          y2partition = Y2Storage::Partition.find_by_name(@device_graph, partition.name)
+          view = ExpertPartitioner::PartitionTreeView.new(y2partition)
+          pane_term = view.create
           id = "partition:" + partition.name
-          item_for(id, partition.sysfs_name)
+          item_for(id, partition.sysfs_name, pane: pane_term)
```

Comment:
Yes, you are right, we should be passing an `AbstractWidget` instead of a term.

Regarding the eager construction of all views beforehand you are also right, we should not do that. I got there by wanting to couple the view to the tree item and to avoid precisely the kind of ID parsing that you quote here. I believe we can find a middle ground by introducing laziness: A `TreeItem` will not refer to an `AbstractWidget`, but to a lambda that produces the widget.